### ISSUE TITLE
undefined symbol: RB_INTEGER_TYPE_P with io-console 0.5.0

### ISSUE THREAD

**Issue by LarsKollstedt** [ruby/io-console#7]
Hi,

after some gem libraries (and the removal of /var/lib/gem did not resolve the following):
```
root@nms:~# journalctl -u oxidized
[...]
Dec 12 13:16:01 nms oxidized[11930]: ssh not found for node <IP>
Dec 12 13:16:01 nms oxidized[11930]: /var/lib/gems/2.3.0/gems/oxidized-0.27.0/lib/oxidized/node.rb:155:in `block in resolve_input': ssh not found for node <IP>
```

Refering to https://github.com/ytti/oxidized/issues/1459 I tried:

```
root@nms:/var/lib/gems/2.3.0/gems# su oxidized 
oxidized@nms:/var/lib/gems/2.3.0/gems$ irb
irb(main):001:0> require "net/ssh"
LoadError: /var/lib/gems/2.3.0/gems/io-console-0.5.0/lib/io/console.so: undefined symbol: RB_INTEGER_TYPE_P - /var/lib/gems/2.3.0/gems/io-console-0.5.0/lib/io/console.so
[...]
```

And then 
```
oxidized@nms:/var/lib/gems/2.3.0/gems$ irb
irb(main):001:0> require "io/console"
LoadError: /var/lib/gems/2.3.0/gems/io-console-0.5.0/lib/io/console.so: undefined symbol: RB_INTEGER_TYPE_P - /var/lib/gems/2.3.0/gems/io-console-0.5.0/lib/io/console.so
```
io-console was also one of the libraries upgraded today. It was upgraded from io-console-0.4.9 to io-console-0.5.0. So this looks to me like a syntax error that came in on that version.

The other changes were:
```
root@nms:/var/lib/gems.bak/2.3.0/gems# ls -la | egrep ' Dec 12'
drwxr-xr-x 150 root root 4096 Dec 12 11:25 .
drwxr-xr-x   4 root root 4096 Dec 12 11:25 io-console-0.5.0
drwxr-xr-x   9 root root 4096 Dec 12 11:25 json-2.3.0
drwxr-xr-x   7 root root 4096 Dec 12 11:25 puma-4.3.1
drwxr-xr-x   5 root root 4096 Dec 12 11:25 rugged-0.28.4.1
```
As far as I can see there was a change from 2019-09-09 adding this line in Commit 634410. Version 0.4.9 dosn't contain this.

Kind regards
     Lars

**Comment by LarsKollstedt** [comment]
Hi,

by the way. Downgrading to 0.4.9 resolved this so far. But that's of course no solution. ;-) Only a workarround to make oxidized work again.

Kind regards,
     Lars

**Comment by hsbt** [comment]
Ruby 2.3 is already EOL status.

**Comment by LarsKollstedt** [comment]
Hi,

you are Requiring Ruby >= 2.2.0 in your gemspec, so at least this dependency is not correct any more. ;-)

By the way ruby 2.3 is the version shipped with Ubuntu 16.4 xenial. I can verifiy this is not happening on Ubuntu 18.4 bionic (with ruby 2.5.0).

The combination of an old Ubuntu LTS with bleeding edge gems might be a bit weird, but such things can happen in some cases.

So I'm requesting you to bump your required ruby version in io-console.gemspec, and hopefully to think of bumping that next time you're relying on new features. ;-)

Kind regards
     Lars
